Hellow World app with server communication

tahir24434 opened the pull request:
Signed-off-by: tahir24434 tahir.rauf1@gmail.com

tahir24434 committed:
Hellow World app with server communication

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

tahir24434 committed:
Update README.md

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

tahir24434 committed:
Merge remote-tracking branch 'upstream/master'

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

sushilks commented on src/hw_test.py:
```diff
@@ -27,6 +33,14 @@ def run_test(self):
+        hwsa.start_test()
+
+        # Call teststartfoo on HAnalyzer class instance
+        hwsa.do_req_resp('helloworld', tout_60s, arg1='AWSOME')
```
Checks are still missing on count.
Rest looks ok.

tahir24434 replied:
Done.

sushilks commented on src/hw_test.py:
```diff
@@ -1,15 +1,21 @@
-
+from sys import path
+path.append("hydra/src/main/python")
 from hydra.lib.runtestbase import RunTestBase
```
change to HydraBase

tahir24434 replied:
Done.

sushilks commented on src/hw_test.py:
```diff
@@ -1,15 +1,21 @@
     def __init__(self):
         self.config = ConfigParser()  # Object contains 'configurations' parsed from hydra.ini config file
-        RunTestBase.__init__(self, test_name='HelloWorld', options=None, app_dirs=['src'])
+        RunTestBase.__init__(self, test_name='HelloWorld', options=None, app_dirs=['src', 'target', 'hydra'])
```
Change to HydraBase

tahir24434 replied:
Done.

tahir24434 committed:
Pub/Sub HelloWorld

1. Test will send 'sendmsg' signal with argument 'x' to publisher.
2. Publisher on recieving the signal will send 'x' number of messages to subsciber.
3. Test will send 'getstats' signal to subscriber.
4. Subscriber will respond with number 'x' (number of messages received by subscriber from publisher)

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

sushilks commented on src/hw_test.py:
```diff
@@ -1,68 +1,102 @@
-        self.hw_client_task_port = None
-        self.add_appid(self.hw_server_app_id)
-        self.add_appid(self.hw_client_app_id)
+        RunTestBase.__init__(self, test_name='HelloWorld', options=None, app_dirs=['src', 'hydra'])
```
change to HydraBase

tahir24434 replied:
Done.

sushilks commented on src/hw_test.py:
```diff
@@ -1,68 +1,102 @@
         r.run_test()
-        time.sleep(60)
+        print ("Communicating sendmsg signal to pub")
+        r.hwpa.do_req_resp('sendmsg', tout_60s, arg1='3')
```
why not arg1=3 
why does it need a string ?

tahir24434 replied:
Done.

sushilks commented on src/hw_test.py:
```diff
@@ -1,68 +1,102 @@
+        print ("Communicating sendmsg signal to pub")
+        r.hwpa.do_req_resp('sendmsg', tout_60s, arg1='3')
+        print ("Communicating getstats signal to pub")
+        time.sleep(50)
```
sleep is not good please remove.

tahir24434 replied:
Removed.

sushilks commented on src/hw_test.py:
```diff
@@ -1,68 +1,102 @@
+        r.hwpa.do_req_resp('sendmsg', tout_60s, arg1='3')
+        print ("Communicating getstats signal to pub")
+        time.sleep(50)
+        r.hwsa.do_req_resp('getstats', tout_60s)
```
instead do a repeated poll every second with a timeout of 60 second.

sushilks replied:
Also the value returned by 'getstats' is not being checked.

tahir24434 replied:
Done.

sushilks commented on src/hw_pub.py:
```diff
@@ -0,0 +1,48 @@
+    pub_rep_port = os.environ.get('PORT0')
+    print ("Starting HelloWorld pub at port [%s]", pub_rep_port)
+    hd = HDHelloWorldPub(pub_rep_port)
+    hd.soc = socket
```
It's generally not a good idea to manipulate a member variable of a class.

sushilks commented on src/hw_pub.py:
```diff
@@ -0,0 +1,48 @@
+        self.register_fn('sendmsg', self.send_msg)
+
+    def send_msg(self, arg1):
+        l.info("send_msg has been called with argument %s" % arg1)
```
arg1 is unused.

tahir24434 replied:
Removed.
